dvc/remote/s3.py

```diff
@@ -187,7 +187,9 @@ def _list_paths(self, path_info, max_items=None):
         )
 
     def walk_files(self, path_info, **kwargs):
-        for fname in self._list_paths(path_info / "", **kwargs):
+        if not kwargs.pop("prefix", False):
+            path_info = path_info / ""
```

Comment:
@efiop after this change we'll treat paths as directories and append the trailing slash for S3-like remote `walk_files` unless `prefix` kwarg is enabled